cgsmiles/read_fragments.py

```diff
@@ -6,6 +6,35 @@
 import pysmiles
 from .read_cgsmiles import read_cgsmiles
 
+class PeekIter(object):
+    """
+    Custom iter that allows looking ahead, without
+    advancing the actual iter.
+    """
+    def __init__(self, collection):
+        self.collection = collection
+        self.index = 0
+
+    def __next__(self):
+        try:
+            result = self.collection[self.index]
+            self.index += 1
+        except IndexError:
+            raise StopIteration
+        return result
+
+    def peek(self):
+        try:
+            result = self.collection[self.index]
+        except IndexError:
+            return ""
+        return result
+
+    def __iter__(self):
+        self.index = 0
+        return self
```

Comment:
You currently need a Sequence that is indexable, rather than just an Iterable.
```suggestion
class PeekIter:
    """
    Custom iter that allows looking ahead, without
    advancing the actual iter.
    """
    def __init__(self, collection):
        self.collection = iter(collection)
        self._peek = None

    def __next__(self):
        item = self._peek or next(self.collection)
        self._peek = None
        return result

    def peek(self):
        if self._peek:
            return self._peek
        self._peek = next(self)
        return self._peek

    def __iter__(self):
        return self
```

Still not perfect, the peeking breaks if you pass it an iterable that contains False-ish items